inc/impact.class.php

```diff
@@ -877,14 +877,19 @@ public static function searchAsset(
             'NOT' => [
                "$table.id" => $used
             ],
-         ] + getEntitiesRestrictCriteria($itemtype::getTable()),
+         ],
       ];
 
       if (is_subclass_of($itemtype, "ExtraVisibilityCriteria", true)) {
          $base_request = array_merge_recursive(
             $base_request,
             $itemtype::getVisibilityCriteria()
          );
+      } else {
```

Comment:
We should indeed check if the itemtype has an entities_id field before adding the result of `getEntitiesRestrictCriteria`.

```suggestion
      }

      $item = new $itemtype();
      if ($item->isEntityAssign()) {
```